portal-kernel/src/com/liferay/document/library/kernel/store/StoreAreaAwareStoreWrapper.java

```diff
@@ -231,6 +232,20 @@ private boolean _isStoreAreaSupported() {
 		return false;
 	}
 
+	private void _registerDeleteDirectoryCallback(
+		long companyId, long repositoryId, String dirName, Store store) {
+
+		TransactionCommitCallbackUtil.registerCallback(
+			() -> {
+				StoreArea.runWithStoreAreas(
+					() -> store.deleteDirectory(
+						companyId, repositoryId, dirName),
+					StoreArea.LIVE, StoreArea.NEW);
+
+				return null;
+			});
+	}
+
```

Comment:
Just a clarification: the PR title says "add background ...", but transactional callbacks are not executed in a different thread; for that you need to create an async message destination/listener. Do you have additional changes in the backburner?

Reply:
no . I will change the title of the Task

Reply:
Ok. I can work on those changes if you want so that you can focus on the other pending stuff.

Reply:
Ok, If you want to work on this PR, feel free and I will not change the name :D
if not. I will change it.

Reply:
Yes, I'll start working on this tomorrow. Setting as draft to avoid anyone forwarding this.